File: src/main/java/io/github/richardson/gameplay/Camera.java
package io.github.richardson.gameplay;

import io.github.richardson.menu.Settings;
import processing.core.PApplet;

public class Camera {
    private final PApplet p;
    private Settings settings;
    private float pos, velocity;

    public Camera(PApplet p, Settings settings, float startPos) {
        this.p = p;
        this.settings = settings;
        this.pos = startPos;
    }

    public void move(float location) {
        velocity = pos + location - p.width * settings.getCameraFocus();
        pos -= velocity * settings.getCameraSmoothness();
        p.translate(pos, p.height);
        p.scale(1, -1);
    }
}

File: src/gameplay/Camera.kt
package gameplay

import menu.Settings
import p

class Camera(private val settings: Settings, var pos: Double) {
    fun move(location: Double) {
        pos += (location - settings.cameraFocus - pos) * settings.cameraSmoothness
        p.scale(settings.zoom, -settings.zoom)
        p.translate((-pos * p.width).toFloat(), (p.height / settings.zoom * -0.9f))
    }
}